prefix-hex/src/data.rs

```diff
@@ -0,0 +1,52 @@
+// Copyright 2022 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use alloc::{format, string::String, vec::Vec};
+
+use crate::{strip_prefix, Error, FromHexPrefix, ToHexPrefix};
+
+impl FromHexPrefix for Vec<u8> {
+    fn from_hex_prefix(hex: &str) -> Result<Self, Error> {
+        let hex = strip_prefix(hex)?;
+        hex::decode(hex).map_err(|e| -> Error { e.into() })
```

Comment:
interesting: is specifying the closures return type necessary? Can we not to `.map_err(Into::into)`?

Reply:
I swear I was battling the compiler for so long. Your suggestion just works 😊.